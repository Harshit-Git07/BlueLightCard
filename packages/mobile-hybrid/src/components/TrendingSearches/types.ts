export interface TrendingSearchesProps {
  trendingSearches: SearchTerms[];
  onTermClick: (data: string) => void;
}

export interface SearchTerms {
  id: number;
  text: string;
}
