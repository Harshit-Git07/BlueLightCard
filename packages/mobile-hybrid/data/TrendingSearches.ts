const TrendingSearchesData = [
  {
    id: 811,
    text: 'Debenhams',
  },
  {
    id: 423,
    text: 'JD Sports',
  },
  {
    id: 12566,
    text: 'LOOKFANTASTIC',
  },
  {
    id: 239,
    text: 'Jet2Holidays',
  },
];

export default TrendingSearchesData;
