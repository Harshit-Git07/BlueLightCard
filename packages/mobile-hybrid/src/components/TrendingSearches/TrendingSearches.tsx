import React, { FC } from 'react';
import { TrendingSearchesProps } from './types';
import { faSearch } from '@fortawesome/pro-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Heading from '@bluelightcard/shared-ui/components/Heading';

const TrendingSearches: FC<TrendingSearchesProps> = ({ trendingSearches, onTermClick }) => {
  return (
    <div className="ml-2">
      <Heading
        headingLevel="h2"
        className="mb-2 text-base text-neutral-grey-900 dark:text-neutral-white pl-3 font-bold"
      >
        Trending searches
      </Heading>
      {trendingSearches.map((search) => (
        <div key={search.id}>
          <button
            className="pl-3 py-3 dark:text-neutral-white text-base w-full h-full text-left"
            onClick={() => onTermClick(search.text)}
          >
            <FontAwesomeIcon
              icon={faSearch}
              size="lg"
              className="pr-3 text-primary-dukeblue-700 dark:text-colour-primary-dark"
              aria-hidden="true"
            />
            {search.text}
          </button>
        </div>
      ))}
    </div>
  );
};

export default TrendingSearches;
