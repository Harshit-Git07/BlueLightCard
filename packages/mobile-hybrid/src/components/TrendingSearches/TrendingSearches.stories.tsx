import TrendingSearches from './TrendingSearches';
import { StoryFn, Meta } from '@storybook/react';
import TrendingSearchesData from '@/data/TrendingSearches';

const componentMeta: Meta<typeof TrendingSearches> = {
  title: 'TrendingSearches',
  component: TrendingSearches,
  parameters: {
    layout: 'fullscreen',
  },
};

const DefaultTemplate: StoryFn<typeof TrendingSearches> = (args) => <TrendingSearches {...args} />;

export const Default = DefaultTemplate.bind({});

Default.args = {
  trendingSearches: TrendingSearchesData,
};

export default componentMeta;
