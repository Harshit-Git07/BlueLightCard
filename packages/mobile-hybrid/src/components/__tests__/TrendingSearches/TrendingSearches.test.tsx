import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TrendingSearches from '@/components/TrendingSearches/TrendingSearches';
import TrendingSearchesData from '@/data/TrendingSearches';

describe('TrendingSearches', () => {
  const mockProps = {
    trendingSearches: TrendingSearchesData,
    onTermClick: jest.fn(),
  };

  it('renders the component', () => {
    const { container } = render(<TrendingSearches {...mockProps} />);

    expect(container).toMatchSnapshot();
  });

  it('registers a user has interacted with a button', () => {
    const handleClick = jest.fn();
    const { getByText } = render(<TrendingSearches {...mockProps} onTermClick={handleClick} />);

    const button = getByText(/LOOKFANTASTIC/i);
    fireEvent.click(button);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
