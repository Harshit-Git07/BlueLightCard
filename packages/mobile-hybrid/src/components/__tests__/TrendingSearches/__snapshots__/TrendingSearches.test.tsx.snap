// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendingSearches renders the component 1`] = `
<div>
  <div
    class="ml-2"
  >
    <h2
      class="text-2xl text-heading-colour dark:text-heading-colour-dark mb-2 font-typography-title-large font-typography-title-large-weight text-typography-title-large leading-typography-title-large tracking-typography-title-large text-colour-onSurface-light dark:text-colour-onSurface-dark
    tablet:font-typography-headline-weight tablet:font-typography-headline tablet:text-typography-headline tablet:tracking-typography-headline tablet:leading-typography-headline mb-2 text-base text-neutral-grey-900 dark:text-neutral-white pl-3 font-bold"
    >
      Trending searches
    </h2>
    <div>
      <button
        class="pl-3 py-3 dark:text-neutral-white text-base w-full h-full text-left"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-magnifying-glass fa-lg pr-3 text-primary-dukeblue-700 dark:text-colour-primary-dark"
          data-icon="magnifying-glass"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
            fill="currentColor"
          />
        </svg>
        Debenhams
      </button>
    </div>
    <div>
      <button
        class="pl-3 py-3 dark:text-neutral-white text-base w-full h-full text-left"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-magnifying-glass fa-lg pr-3 text-primary-dukeblue-700 dark:text-colour-primary-dark"
          data-icon="magnifying-glass"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
            fill="currentColor"
          />
        </svg>
        JD Sports
      </button>
    </div>
    <div>
      <button
        class="pl-3 py-3 dark:text-neutral-white text-base w-full h-full text-left"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-magnifying-glass fa-lg pr-3 text-primary-dukeblue-700 dark:text-colour-primary-dark"
          data-icon="magnifying-glass"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
            fill="currentColor"
          />
        </svg>
        LOOKFANTASTIC
      </button>
    </div>
    <div>
      <button
        class="pl-3 py-3 dark:text-neutral-white text-base w-full h-full text-left"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-magnifying-glass fa-lg pr-3 text-primary-dukeblue-700 dark:text-colour-primary-dark"
          data-icon="magnifying-glass"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
            fill="currentColor"
          />
        </svg>
        Jet2Holidays
      </button>
    </div>
  </div>
</div>
`;
